gazebo/common/HeightmapData.cc

```diff
@@ -95,6 +112,41 @@ HeightmapData *HeightmapDataLoader::LoadTerrainFile(
     return LoadDEMAsTerrain(_filename);
   }
 }
+
+//////////////////////////////////////////////////
+/// Load terrain files taking into account the spherical
+/// coordinates surface type.
+HeightmapData *HeightmapDataLoader::LoadTerrainFile(
+    const std::string &_filename,
+    common::SphericalCoordinatesPtr _sphericalCoordinates)
+{
+  // Register the GDAL drivers
+  GDALAllRegister();
+
+  GDALDataset *poDataset = reinterpret_cast<GDALDataset *>
+      (GDALOpen(_filename.c_str(), GA_ReadOnly));
+
+  if (!poDataset)
+  {
+    gzerr << "Unrecognized terrain format in file [" << _filename << "]\n";
+    return nullptr;
+  }
+
+  std::string fileFormat = poDataset->GetDriver()->GetDescription();
+  GDALClose(reinterpret_cast<GDALDataset *>(poDataset));
+
+  // Check if the heightmap file is an image
+  if (fileFormat == "JPEG" || fileFormat == "PNG")
+  {
+    // Load the terrain file as an image
+    return LoadImageAsTerrain(_filename);
+  }
+  else
+  {
+    // Load the terrain file as a DEM
+    return LoadDEMAsTerrain(_filename, _sphericalCoordinates);
```

Comment:
this new function is almost identical to the existing overload. I'd recommend avoiding code duplication by calling one method from the other

Reply:
Done : https://github.com/osrf/gazebo/pull/3258/commits/26e9bd775983270523aef5517f77e2711bb1ed25